pkg/action/archive.go

```diff
@@ -0,0 +1,206 @@
+package action
+
+import (
+	"archive/tar"
+	"archive/zip"
+	"compress/gzip"
+	"errors"
+	"fmt"
+	"io"
+	"log"
+	"os"
+	"path"
+	"path/filepath"
+	"strings"
+
+	"github.com/ulikunitz/xz"
+)
+
+const maxBytes = 1 << 29 // 512MB
```

Comment:
This seems like an acceptably high archive size. If anything, it can probably be lower.